### PR TITLE
Optionally load evil-integration.el

### PR DESCRIPTION
**Pull request by edkolev**
This is a WIP PR as a result of #992 

I haven't self-tested thoroughly yet. Plus, there's an issue with the generation of `.depend` which is currently worked-around like so:

```elisp
(when evil-want-integration
;; TODO creating the .depend file breaks if the following require is
;; not on a line of its own
(require 'evil-integration)
)
```

Edit: the workaround is is now removed.

Any suggestions for improvements are welcome.

### CODE REVIEW

**Comment by noctuid** [comment]
I think that it would be much preferable to split the files into one with keybindings and one with everything else (or to further split everything else, but I think the main thing most people would want to disable is only the keybindings). See my comment in #992. I don't think everyone would agree on what is "basic" (for example, I'd include the advice and setting of evil command properties).

**Comment by wasamasa** [comment]
Sigh. I don't want to split up anything at all.

**Comment by VanLaser** [comment]
For one, yes, I'd like evil not to load these keybindings by default, so I'd welcome a split too. Right now I have to use (are these easy to find for an evil newcomer? I think no):

```elisp
  ;; don't let modes override any states (!)
  (setq evil-overriding-maps nil
        evil-intercept-maps nil
        evil-pending-intercept-maps nil
        evil-pending-overriding-maps nil)
  ;; subvert evil-operation.el overrides (dired, ibuffer etc.)
  (advice-add 'evil-make-overriding-map :override #'ignore)
  (advice-add 'evil-make-intercept-map  :override #'ignore)
  (advice-add 'evil-add-hjkl-bindings   :override #'ignore)
```

... so it would be much more "natural" to not have to retort to these workarounds.

**Comment by TheBB** [comment]
I'm fine making evil-integration optional. Strongly agree with @wasamasa, let's not split it.

**Comment by noctuid** [comment]
Providing a setting to not load `evil-integration` without splitting it doesn't really solve anything. Users still have to require it for the configuration they want and use workarounds to prevent/undo the keybindings/overriding maps parts. That or they have to not load it all (now with an official setting) and copy the other parts into their own config or a new package. Nothing really changes.

Why don't you want to split it?

**Comment by wasamasa** [comment]
See the linked ticket.

**Comment by edkolev** [comment]
OK, this PR is now modified - it only adds the option to load evil-integration.el optionally. The generation of the .depend file is also tweaked.

File: Makefile [hunk 1]
```diff
@@ -20,7 +20,7 @@ compile: $(ELCFILES)
 	@echo Compute dependencies
 	@rm -f .depend
 	@for f in $(FILES); do \
-	    sed -n "s/(require '\(evil-.*\))/$${f}c: \1.elc/p" $$f >> .depend;\
+	    sed -n "s/[^(]*(require '\(evil-[^)]*\).*)/$${f}c: \1.elc/p" $$f >> .depend;\
```
Comment (wasamasa) [inline Makefile]: This seems somewhat broad, can't you just allow leading space in front of `(require 'evil-...)`?

**Comment by edkolev** [comment]
Sure, the commit is now amended - only spaces are allowed in front on the `(require ...`

**Comment by wasamasa** [comment]
Thanks!

**Pull request merged**